.github/workflows/docs.yaml

```
@@ -23,8 +23,46 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
+          fetch-depth: 0 # needed for checking changes and pushing
           submodules: recursive
 
+      - name: Set up Python
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      - name: Install Poetry
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

```suggestion
        if: github.event_name == 'push'
```

You've already restricted `push` to the `main` branch in the workflow triggers at the top of the file.

---

The same also applies below.